### DISCUSSION TITLE
Missing usage examples?

### DISCUSSION THREAD

**Discussion by javier-sierra-sngular** [node-saml/node-saml#359] [Q&A] [answered]
Sorry, guys, but I think I am missing something...

I noticed this in the README:
> The examples utilize the [Feide OpenIdp identity provider](https://openidp.feide.no/) [...]
But I could not find any usage examples and the API of SAML object.

However, I couldn't find any usage examples or documentation on the SAML object's API.

I'm attempting to use this library instead of passport-saml because I don't need sessions and other features provided by passport. I simply need a straightforward SAML authentication process, but I'm unsure how to use this library.

Could you please point me in the right direction? Thank you.

**Comment by srd90** [comment, marked as answer]
That feide thing etc. exists at node-saml's readme because once up on the time there was only `passport-saml` which was later splitted to passport specific part and to saml specific part. Content of the node-saml's readme still contains stuff from the past.

>...I simply need a straightforward SAML authentication process, but I'm unsure how to use this library.

`@node-saml/passport-saml` is one example. It is passport strategy which uses `@node-saml/node-saml` to process responses from IdP/generate requests to IdP (see `passport-saml`'s [strategy.ts](https://github.com/node-saml/passport-saml/blob/v4.0.4/src/strategy.ts)).

Related threads/issues:
* https://github.com/node-saml/node-saml/discussions/42
* https://github.com/node-saml/node-saml/issues/258

BTW. "Straightforward SAML authentication process" is somewhat vague question. Roots of node-saml library is at the library which implements SAML web SSO profile (passport-saml). If your use case is to implement web sso profile with some other technology than passport you have to know how to use some other technology's APIs/authebtication framework/infra and you have studied SAML specs e.g. about web SSO profile. Once you have done preliminary studies then public method names of https://github.com/node-saml/node-saml/blob/v4.0.5/src/saml.ts are obvious. It is very hard for anyone to provide any answers for your question because you do not provide any hint whats your vision  (and technologies to be used) of "straithforward saml authentication process".

Here is links to SAML specifications: https://github.com/node-saml/node-saml/issues/326#issuecomment-1717221811
